pkgs/development/libraries/glm/default.nix

```diff
@@ -1,37 +1,55 @@
-{ lib, stdenv, fetchurl, fetchzip, cmake }:
+{ lib, stdenv, fetchFromGitHub, cmake }:
 
 stdenv.mkDerivation rec {
-  version = "0.9.8.5";
+  version = "0.9.9.8";
   pname = "glm";
 
-  src = fetchzip {
-    url = "https://github.com/g-truc/glm/releases/download/${version}/${pname}-${version}.zip";
-    sha256 = "0dkfj4hin3am9fxgcvwr5gj0h9y52x7wa03lfwb3q0bvaj1rsly2";
+  src = fetchFromGitHub {
+    owner = "g-truc";
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-F//+3L5Ozrw6s7t4LrcUmO7sN30ZSESdrPAYX57zgr8=";
   };
 
+  outputs = [ "out" "doc" ];
+
   nativeBuildInputs = [ cmake ];
 
-  outputs = [ "out" "doc" ];
+  cmakeFlags = [
+    "-DBUILD_SHARED_LIBS=OFF"
+    "-DBUILD_STATIC_LIBS=OFF"
+    "-DGLM_TEST_ENABLE=${if doCheck then "ON" else "OFF"}"
+  ];
 
-  cmakeConfigureFlags = [ "-DGLM_INSTALL_ENABLE=off" ];
+  doCheck = true;
 
-  # fetch newer version of platform.h which correctly supports gcc 7.3
-  gcc7PlatformPatch = fetchurl {
-    url = "https://raw.githubusercontent.com/g-truc/glm/384dab02e45a8ad3c1a3fa0906e0d5682c5b27b9/glm/simd/platform.h";
-    sha256 = "0ym0sgwznxhfyi014xs55x3ql7r65fjs34sqb5jiaffkdhkqgzia";
-  };
+  installPhase = let
+    sourceDir = "$NIX_BUILD_TOP/$sourceRoot";
+  in ''
```

Comment:
We should use a relative directory instead.

Reply:
I thought about it and just use `..` (because the working directory when using CMake is `build`), but the previous version of the derivation was doing it like that, and I thought that maybe I should not change it. I also checked that several other derivations use the same `NIX_BUILD_TOP` environment variable, that's why I kept it.

```
$ rg --files-with-matches  NIX_BUILD_TOP pkgs | wc -l
      68
```

Reply:
Updated, and rebased on top of current HEAD of master.